### PR TITLE
Creating extra geographic look-ups

### PR DESCRIPTION
**Pull request by rmbielby**
Adding the following geographic lookups to the screener:
- LA - PCon lookup sourced from [ONS tables 2017-2023](https://geoportal.statistics.gov.uk/search?q=LUP_WPC&sort=Title%7Ctitle%7Casc&tags=PCON)
- LSIP - LAD lookup sourced from [ONS tables 2023](https://geoportal.statistics.gov.uk/datasets/ons::lad-to-local-skills-improvement-plan-areas-august-2023-lookup-in-england/about)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 .Rhistory
 .RData
 .Ruserdata
-README.html
\ No newline at end of file
+README.html
+data/downloaded_source_data/
```
Comment (cjrace) [inline .gitignore:6]: Worth adding a note in the README to explain this folder and pointing to the R script for it?

File: R/standard-data-prep/update-geography-lookups.R [hunk 2]
```diff
@@ -0,0 +1,98 @@
+library(readr)
+# @param open_geography_file Lookup file sourced from
```
Comment (cjrace) [inline R/standard-data-prep/update-geography-lookups.R:3]: Is this intended as roxygen or just our own version? I think roxygen2 would need a ' after the # to work

File: R/standard-data-prep/update-geography-lookups.R [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+#       https://geoportal.statistics.gov.uk/search?q=LUP_WPC&sort=Date%20Created%7Ccreated%7Cdesc
+#       e.g. data/downloaded_source_data/Ward_to_Westminster_Parliamentary_Constituency_to_Local_Authority_District_to_Upper_Tier_L.csv
+update_pcon_la_lookup <- function(open_geography_file, existing_data_file = "data/la_pcon_hierarchy.csv") {
```
Comment (cjrace) [inline R/standard-data-prep/update-geography-lookups.R]: Outputs of this function in the console are quite verbose / makes it miss the message you added in there, could  cut out the bits from read_csv() by using `show_col_types = FALSE`?

image

Then would consider adding a starting message confirming where it's reading the new file from (e.g. 'Reading X from data/downloaded_source_data', and a closing one to say 'Writing updated data/la_pcon_hierarchy.csv file'?

File: R/standard-data-prep/update-geography-lookups.R [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  }
+  new_lookup %>%
+    arrange(pcon_code, new_la_code) %>%
+    filter(grepl("E", pcon_code)) %>% # Filter to only include England
+    write.csv("data/la_pcon_hierarchy.csv", row.names = FALSE)
+}
+
+# @param open_geography_file Lookup file sourced from
+#       https://geoportal.statistics.gov.uk/search?q=lsip&sort=Title%7Ctitle%7Casc
+update_lad_lsip_lookup <- function(open_geography_file, existing_data_file = "data/lsip_lad_hierarchy.csv") {
```
Comment (cjrace) [inline R/standard-data-prep/update-geography-lookups.R]: Would suggest similar actions here to what I did above around the messaging in the console, it's a minor thing but figure it's probably fairly quick to tidy up:

image

File: R/standard-data-prep/update-geography-lookups.R [hunk 5]
```diff
@@ -0,0 +1,98 @@
+library(readr)
```
Comment (cjrace) [inline R/standard-data-prep/update-geography-lookups.R:1]: Tried running it in isolation from the app and you also need library(dplyr) here too

**Review by cjrace — CHANGES_REQUESTED** [review]

File: R/standard-data-prep/update-geography-lookups.R (hunk 2 above)
Comment (rmbielby) [reply R/standard-data-prep/update-geography-lookups.R:3]: Was just using the roxygen's @param terminology to make a note really. Figured it wasn't worth doing full roxygen, but worth stealing some of the syntax! Happy to add the ' if you think we might start using roxygen in this repo.

File: R/standard-data-prep/update-geography-lookups.R (hunk 3 above)
Comment (rmbielby) [reply R/standard-data-prep/update-geography-lookups.R]: Done

File: R/standard-data-prep/update-geography-lookups.R (hunk 4 above)
Comment (rmbielby) [reply R/standard-data-prep/update-geography-lookups.R]: Done

File: R/standard-data-prep/update-geography-lookups.R (hunk 5 above)
Comment (rmbielby) [reply R/standard-data-prep/update-geography-lookups.R:1]: Done

File: R/standard-data-prep/update-geography-lookups.R (hunk 2 above)
Comment (cjrace) [reply R/standard-data-prep/update-geography-lookups.R:3]: I'm pretty sure we will one day, and then turn this app into a package as well as the app so we can embed it in analysts pipeline but tbh it's fine to leave for now, we can add at later date.

File: .gitignore (hunk 1 above)
Comment (cjrace) [reply .gitignore:6]: Think I've covered this in the latest commit - feel free to correct / change if appropriate!

**Review by cjrace — APPROVED** [review]

**Pull request merged**